zk.js/src/utxo/program-utxo.ts

```diff
@@ -17,75 +15,128 @@ import {
   createAccountObject,
 } from "../utils";
 import {
-  createOutUtxo,
   encryptOutUtxoInternal,
   decryptOutUtxoInternal,
   Utxo,
   CreateUtxoInputs,
-  createUtxo,
   OutUtxo,
+  checkAssetAndAmountIntegrity,
+  getDefaultUtxoTypeAndVersionV0,
+  randomBN,
+  getUtxoHashInputs,
+  getUtxoHash,
+  createNullifierWithAccountSignature,
 } from "./utxo";
 import { Account } from "../account";
+import { BN31, createBN31 } from "./bn31";
 
-// TODO: make utxoData depend on idl with generic type
-export type ProgramOutUtxo = {
-  outUtxo: OutUtxo;
-  pspId: PublicKey;
-  pspIdl: Idl;
-  includeUtxoData: boolean;
-  utxoName: string;
+export type PlaceHolderTData = any;
+/** Program-owned utxo that had previously been inserted into a state Merkle tree */
+export type ProgramUtxo<TData extends PlaceHolderTData> = Omit<
+  Utxo,
+  "owner"
+> & {
+  /** Public key of program that owns the utxo */
+  owner: PublicKey;
+  /** Data assigned to the utxo */
+  data: TData;
+  /** Hash of 'data' */
+  dataHash: BN31;
+  /** psp idl */
+  ownerIdl: Idl; /// TODO: remove from utxo (waste of space)
```

Comment:
agreed in a future pr we could just get it with a function from a mapping with all system psp idls or pass it independently when needed.